asab/pdict.py

```diff
@@ -76,16 +55,15 @@ def update(self, other=(), **kwds):
 		Inspired by a https://github.com/python/cpython/blob/3.8/Lib/_collections_abc.py
 		"""
 
-		with shelve.open(self._path) as d:
-			if isinstance(other, collections.Mapping):
-				for key in other:
-					d[key] = other[key]
-			elif hasattr(other, "keys"):
-				for key in other.keys():
-					d[key] = other[key]
-			else:
-				for key, value in other:
-					d[key] = value
-
-			for key, value in kwds.items():
-				d[key] = value
+		if isinstance(other, collections.Mapping):
+			for key in other:
+				self[key] = other[key]
+		elif hasattr(other, "keys"):
```

Comment:
What is this?

Reply:
As it says in doc-string. This was inspired by
https://github.com/python/cpython/blob/3.8/Lib/_collections_abc.py#L833

Maybe it could be removed since we inherit most of the functionality from `dict`?